app/views/users/index.html.erb

```diff
@@ -1 +1,7 @@
-<h1>User posts</h1>
\ No newline at end of file
+<div>
+  <% @users.each do |user| %>
+    <%= link_to user_path(user.id) do %>
+      <%= render partial: "user_profile", locals: {user: user} %>
+    <% end %>
+  <% end %>
```

Comment:
- [ ] [OPTIONAL] this is great. as an improvement, you can always avoid creating an iteration loop because the `render` function can render an entire collection if you pass it in.
```suggestion
  <%= render collection: @users, partial: "user_profile", as: :user, layout: 'link' %>
```
In this case you can also add a layout partial that will wrap each user so the link still works

```erb
// _link.html.erb
<%= link_to user_path(user) do %>
  <%= yield %>
<% end %>
```